demo/lightning/checkpoint/simulated/multiprocessing_train.py

```diff
@@ -0,0 +1,190 @@
+    parser.add_argument("--project", type=str,
+                        required=True, help="GCS project ID.")
+    parser.add_argument("--ckpt-dir-path", type=str, required=True,
+                        help="Path to GCS bucket for checkpoints.")
+    parser.add_argument("--model-size", type=int,
+                        default=100, help="Size of the model.")
+    parser.add_argument("--clear-kernel-cache", action="store_true",
+                        default=False, help="Clear kernel cache.")
+    parser.add_argument("--sample-count", type=int, default=3,
+                        help="Number of samples for benchmarking.")
+    parser.add_argument("--padding-size", type=int, default=1000,
+                        help="Size of dummy tensors for padding, to control the size of the checkpoint.")
+    parser.add_argument("--world-size", type=int, required=True,
+                        help="Number of processes in the distributed setup.")
+    return parser.parse_args()
+
+
+def format_size(size_bytes: int) -> str:
+    """Formats bytes into a human-readable size string."""
+    size_mb = size_bytes / (1024 * 1024)
+    return f"{size_mb / 1024:.2f} GB" if size_mb >= 1024 else f"{size_mb:.2f} MB"
+
+
+def get_tensor_size_bytes(tensor: torch.Tensor) -> int:
+    """Calculates the size of a tensor in bytes."""
+    return tensor.element_size() * tensor.numel()
+
+
+def setup(rank: int, world_size: int) -> None:
+    """Sets up the distributed environment."""
+    os.environ['MASTER_ADDR'] = MASTER_ADDR
+    os.environ['MASTER_PORT'] = MASTER_PORT
+    dist.init_process_group("gloo", rank=rank, world_size=world_size)
+
+
+def cleanup() -> None:
+    """Cleans up the distributed environment."""
+    dist.destroy_process_group()
+
+
+def split_tensor(tensor: torch.Tensor, world_size: int, rank: int) -> torch.Tensor:
+    """Splits a tensor into chunks for distributed processing."""
+    numel = tensor.numel()
+    split_size = numel // world_size
+    start_idx = rank * split_size
+    end_idx = start_idx + split_size if rank != world_size - 1 else numel
+    return tensor.view(-1)[start_idx:end_idx]
+
+
+def time_checkpoint_operation(benchmark_strategy: BenchmarkStrategy, distributed_state_dict: Dict[str, torch.Tensor], filepath: str, sample_count: int, operation: str) -> list:
+    """Times save or load operations for checkpoints."""
+    times = []
+    for i in range(sample_count):
+        checkpoint_path = os.path.join(
+            filepath, f'checkpoints/ckpt_{i}.ckpt')
+        start_time = time.time()
+        if operation == 'save':
+            benchmark_strategy.save_checkpoint(
+                distributed_state_dict, filepath=checkpoint_path)
+        elif operation == 'load':
+            benchmark_strategy.load_checkpoint(checkpoint_path=checkpoint_path)
+        end_time = time.time()
+        times.append(end_time - start_time)
+        dist.barrier()  # Synchronize processes
```

Comment:
You might want to print the save/load time here from rank 0. When saves/loads take a long time, this will help the user have confidence that it is making progress.

Reply:
Good point. Will do !